src/utils/CoinbasePixel.ts

```diff
@@ -117,71 +120,83 @@ export class CoinbasePixel {
-          this.startDirectSignin(openEmbeddedExperience);
+    if (experience === 'embedded') {
+      const openEmbeddedExperience = () => {
+        const embedded = createEmbeddedContent({ url, ...embeddedContentStyles });
+        if (embeddedContentStyles?.target) {
+          document.querySelector(embeddedContentStyles?.target)?.replaceChildren(embedded);
         } else {
-          openEmbeddedExperience();
+          document.body.appendChild(embedded);
         }
-      } else if (experience === 'popup' && window.chrome?.windows?.create) {
-        void window.chrome.windows.create(
-          {
-            url,
-            setSelfAsOpener: true,
-            type: 'popup',
-            focused: true,
-            width: PopupSizes.signin.width,
-            height: PopupSizes.signin.height,
-            left: window.screenLeft - PopupSizes.signin.width - 10,
-            top: window.screenTop,
-          },
-          (winRef) => {
-            this.addEventStreamListener('open', () => {
-              if (winRef?.id) {
-                chrome.windows.update(winRef.id, {
-                  width: PopupSizes.widget.width,
-                  height: PopupSizes.widget.height,
-                  left: window.screenLeft - PopupSizes.widget.width - 10,
-                  top: window.screenTop,
-                });
-              }
-            });
-          },
-        );
-      } else if (experience === 'new_tab' && window.chrome?.tabs?.create) {
-        void window.chrome.tabs.create({ url });
+      };
+
+      if (!this.isLoggedIn) {
+        // Embedded experience opens popup for signin
+        this.startDirectSignin(openEmbeddedExperience);
       } else {
-        openWindow(url, experience);
+        openEmbeddedExperience();
       }
-    });
+    } else if (experience === 'popup' && window.chrome?.windows?.create) {
+      void window.chrome.windows.create(
+        {
+          url,
+          setSelfAsOpener: true,
+          type: 'popup',
+          focused: true,
+          width: PopupSizes.signin.width,
+          height: PopupSizes.signin.height,
+          left: window.screenLeft - PopupSizes.signin.width - 10,
+          top: window.screenTop,
+        },
+        (winRef) => {
+          this.addEventStreamListener('open', () => {
+            if (winRef?.id) {
+              chrome.windows.update(winRef.id, {
+                width: PopupSizes.widget.width,
+                height: PopupSizes.widget.height,
+                left: window.screenLeft - PopupSizes.widget.width - 10,
+                top: window.screenTop,
+              });
+            }
+          });
+        },
+      );
+    } else if (experience === 'new_tab' && window.chrome?.tabs?.create) {
+      void window.chrome.tabs.create({ url });
+    } else {
+      openWindow(url, experience);
+    }
+
+    // Add an event listener for when the widget opens so we can set a new nonce without invalidating the current nonce
```

Comment:
@lachiet Can we get a comment with a bit more context on why we need two nonces at once?